libpod/kube.go

```diff
@@ -505,9 +502,34 @@ func libpodMountsToKubeVolumeMounts(c *Container) ([]v1.VolumeMount, []v1.Volume
 		vms = append(vms, vm)
 		vos = append(vos, vo)
 	}
+	for _, v := range namedVolumes {
+		vm, vo := generateKubePersistentVolumeClaim(v)
+		vms = append(vms, vm)
+		vos = append(vos, vo)
+	}
 	return vms, vos, nil
 }
 
+// generateKubePersistentVolumeClaim converts a ContainerNamedVolume to a Kubernetes PersistentVolumeClaim
+func generateKubePersistentVolumeClaim(v *ContainerNamedVolume) (v1.VolumeMount, v1.Volume) {
+	ro := util.StringInSlice("ro", v.Options)
+
+	// To avoid naming conflicts with any host path mounts, add a unique suffix to the volume's name.
+	name := v.Name + "-pvc"
+
+	vm := v1.VolumeMount{}
+	vm.Name = name
+	vm.MountPath = v.Dest
+	vm.ReadOnly = ro
+
+	pvc := v1.PersistentVolumeClaimVolumeSource{ClaimName: v.Name, ReadOnly: ro}
+	vs := v1.VolumeSource{}
+	vs.PersistentVolumeClaim = &pvc
+	vo := v1.Volume{Name: name, VolumeSource: vs}
+
+	return vm, vo
+}
+
```

Comment:
Are there man pages changes necessary to at least note this new ability?

Reply:
There's barely any mention at all related to volumes in either the podman-play-kube(1) or podman-generate-kube(1) man pages.
The corresponding change to add PersistentVolumeClaims support to podman-play-kube didn't update the man pages.
This would probably require creating sections describing how Podman handles volumes for both podman-play-kube(1) and podman-generate-kube(1).
It's definitely something I would have found helpful to have documentation on.
If that's considered within the scope of this PR, I can write-up those sections.

Reply:
@jwillikers I'd really like to see the info there in the man pages about the volumes.  It would be awesome if you could provide it.  That said, if you want to create an issue to that effect, then I'm fine merging this in now, or you could add the examples to this PR.  I'll let you make the call.  

If you go the issue route, please add a link to it in a comment here.  Thanks!

Reply:
Thanks for getting back to me on that. I don't have a problem adding it to this PR, I should be able to take care of that later this afternoon. 👍

Reply:
@TomSweeneyRedHat I've updated the documentation for podman-play-kube(1) and podman-generate-kube(1).
However, in doing so it brought up the fact that there are now multiple ways to specify named volumes with `podman play kube`, through a relative path using the `hostPath` volume type _or_ a `persistentVolumeClaim` volume type. I documented both, but wasn't sure if the documentation should prefer one method over the other.